marlowe-dashboard-client/src/MainFrame/State.purs

```diff
@@ -253,7 +265,7 @@ updateRunningContracts companionState = do
                   modifying (_playState <<< _allContracts) $ insert plutusAppId contractState
                   handleAction $ PlayAction $ Play.ContractHomeAction $ ContractHome.OpenContract plutusAppId
                   addToast $ successToast "You have been given a role in a new contract."
-                Nothing -> addToast $ errorToast "Could not determine contract type." $ Just "You have been given a role in a new contract, but we could not determine the type of the contract and therefore cannot display it. This should not happen. Please contact support."
+                Nothing -> addToast $ errorToast "Could not determine contract type." $ Just "You have been given a role in a new contract, but we could not determine the type of the contract and therefore cannot display it."
```

Comment:
I think this is information for the developer, if we do things right this message should never see the light of day. I'd put either `Nothing` or something like 

"We reached an unexpected situation, please contact support and provide this id XXXX"

Reply:
I feel the longer explanation is necessary, because this error (if it did ever happen - which of course it shouldn't) could just pop up randomly at any time. With the extra information of "You have been given a role in a new contract...", the user can then see why this suddenly appeared out of nowhere.